src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -398,5 +392,70 @@ private static Version SafeParseVersion(string version)
                 return new Version(0, 0);
             }
         }
+
+        
+        /// <summary>
+        /// This class is used to create a temp file, which is deleted in Dispose().
+        /// </summary>
+        private class TempFile : IDisposable
+        {
+            private readonly string _filePath;
+
+            /// <summary>
+            /// Constructor. It creates an empty temp file under directory <paramref name="directory"/>, with 
+            /// extension <paramref name="extension"/>.
+            /// </summary>
+            /// <param name="directory">The directory where the temp file is created.</param>
+            /// <param name="extension">The extension of the temp file.</param>
+            public TempFile(string directory, string extension)
+            {
+                if (string.IsNullOrEmpty(directory))
+                {
+                    throw new ArgumentNullException(nameof(directory));
+                }
+
+                if (string.IsNullOrEmpty(extension))
+                {
+                    throw new ArgumentNullException(nameof(extension));
+                }
+
+                int count = 0;
+                do
+                {
+                    _filePath = Path.Combine(directory, Path.GetRandomFileName() + extension);
+                    count++;
+                }
+                while (File.Exists(_filePath) && count < 3);
+
+                if (count == 3)
+                {
+                    throw new InvalidOperationException("Failed to create a random file.");
+                }
+
+                // create an empty file
+                using (var filestream = File.Open(_filePath, FileMode.CreateNew))
```

Comment:
you probably want to move the file creation inside the loop, so you also guarantee race conditions. Long shot that this would ever happen, but why not